internal/test/builder/builders.go

```diff
@@ -1069,7 +1330,7 @@ func (m *MachinePoolBuilder) WithBootstrapTemplate(ref *unstructured.Unstructure
 	return m
 }
 
-// WithInfrastructureTemplate adds the passed unstructured object to the MachinePool builder as an infrastructureMachineTemplate.
+// WithInfrastructureTemplate adds the passed unstructured object to the MachinePool builder as an infrastructureMachinePoolTemplate.
 func (m *MachinePoolBuilder) WithInfrastructureTemplate(ref *unstructured.Unstructured) *MachinePoolBuilder {
```

Comment:
Let's change this and the description to WithInfrastructure

same for bootstrap. That should be WithBootstrap (in l.1328)

Reply:
Sure. I guess then I should also rename the fields (l. 1310 & 1311) to `bootstrap` and `infrastructure` as well, as they aren't in fact tempaltes like for MDs?!

Reply:
Yup!